server/zio-http-server/src/test/scala/sttp/tapir/server/ziohttp/ZioHttpTestServerInterpreter.scala

```diff
@@ -21,15 +21,17 @@ class ZioHttpTestServerInterpreter(eventLoopGroup: EventLoopGroup, channelFactor
 
   override def server(routes: NonEmptyList[Http[Any, Throwable, Request, Response]]): Resource[IO, Port] = {
     implicit val r: Runtime[Any] = Runtime.default
-    val layers: ZLayer[Any, Nothing, EventLoopGroup with ServerChannelFactory] =
-      ZLayer.succeed(eventLoopGroup) ++ ZLayer.succeed(channelFactory)
 
-    val server: Server[Any, Throwable] = Server.app(routes.toList.reduce(_ ++ _))
+    val io: ZIO[Scope, Throwable, Int] =
+      (for {
+        driver <- ZIO.service[Driver]
+        _ <- driver.start(trace)
+        _ <- driver.addApp[Any](routes.toList.reduce(_ ++ _), ZEnvironment())
+      } yield 0).provideSome[Scope](
```

Comment:
I think you should return the assigned port here - so that the client can make the request; `driver.start` seems to return the port

Reply:
Sorry this was just me being stupid, didn't realize `0` auto allocated a free port 🤦

Reply:
Yeah 0 is correct as we need an auto-assigned random port, but then we need to return the allocated port to the test client :)